File: Vehicle With Polymorphism/Vehicle With Polymorphism/Vehicle.cpp
#include "Vehicle.h"

Vehicle::Vehicle(const std::string& make, const std::string& model, int year) : make(make), model(model), year(year) {}

std::string Vehicle::getMake() const {
	return make;
}

std::string Vehicle::getModel() const {
	return model;
}

int Vehicle::getYear() const {
	return year;
}

File: IGP220-IGP230 - C# and ASP.NET/Assignments/Dog Class/Car (Maya's Code)/Car/Vehicle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Car
{
    internal abstract class Vehicle
    {
        public string Make { get; set; }
        public string Model { get; set; }
        public int Year { get; set; }
        public Vehicle(string make, string model, int year)
        {
            Make = make;
            Model = model;
            Year = year;
        }
        public abstract string Drive();
        public void Start()
        {
            MessageBox.Show("Vehicle started.");
        }
        public void Stop()
        {
            MessageBox.Show("Vehicle stopped.");
        }
    }



}